Tests/test_Affy.py

```diff
@@ -0,0 +1,102 @@
+import unittest
+
+from Bio.Affy import CelFile
+import struct
+from numpy import array
+import numpy.testing
+
+class AffyTest(unittest.TestCase):
+    def setUp(self):
+        self.affy3 = "Affy/affy_v3_example.CEL"
+        self.affy4 = "Affy/affy_v4_example.CEL"
+    def tearDown(self):
+        pass
+    
+    # tests if the new strict mode complains about passing a non-file
+    def testAffyStrict(self):
+        with self.assertRaises(IOError) as context:
+            with open(self.affy3, "rb") as f:
+                record = CelFile.read("hello", strict=True)
+
+    # tests if the new strict mode is backwards compatible
+    def testAffyStrict(self):
+        with open(self.affy3, "rb") as f:
+            record = CelFile.read("hello")
+        assert record.DatHeader is None
+
+    # tests the old Affymetrix v3 parser
+    def testAffy3(self):
+        with open(self.affy3, "r") as f:
+            record = CelFile.read(f)
+            assert(len(record.DatHeader) > 0)
+            assert(record.intensities.shape == (5, 5))
+            assert(record.intensities.shape == record.stdevs.shape)
+            assert(record.intensities.shape == record.npix.shape)
+            assert(record.ncols == 5)
+            assert(record.nrows == 5)
+    # tests the new Affymetrix v4 parser
+    def testAffy4(self):
+        with open(self.affy4, "rb") as f:
+            record = CelFile.read(f)
+            assert(record.intensities.shape == (5, 5))
+            assert(record.intensities.shape == record.stdevs.shape)
+            assert(record.intensities.shape == record.npix.shape)
+            assert(record.ncols == 5)
+            assert(record.nrows == 5)
+            numpy.allclose(record.intensities, [[  0.,   1.,   2.,   3.,   4.],
+                                                [  5.,   6.,   7.,   8.,   9.],
+                                                [ 10.,  11.,  12.,  13.,  14.],
+                                                [ 15.,  16.,  17.,  18.,  19.],
+                                                [ 20.,  21.,  22.,  23.,  24.]])
+            numpy.allclose(record.stdevs, [[  0.,  -1.,  -2.,  -3.,  -4.],
+                                           [ -5.,  -6.,  -7.,  -8.,  -9.],
+                                           [-10., -11., -12., -13., -14.],
+                                           [-15., -16., -17., -18., -19.],
+                                           [-20., -21., -22., -23., -24.]])
+            numpy.allclose(record.npix, [[9, 9, 9, 9, 9],
+                                         [9, 9, 9, 9, 9],
+                                         [9, 9, 9, 9, 9],
+                                         [9, 9, 9, 9, 9],
+                                         [9, 9, 9, 9, 9]])
+
+# Writes a small example Affymetrix V4 CEL File
+def writeExampleV4():
```

Comment:
I moved it to be run every time a testcase is run. Every time a testcase is run the test file is generated anew.